### ISSUE TITLE
您好，请问LSTUR在验证集上auc可以达到多少呢？

### ISSUE THREAD

**Issue by qq31415926** [nnnyt/NewsRecommendation#1]

**Comment by nnnyt** [comment]
你好，在[MIND-small](https://msnews.github.io)上LSTUR-ini和LSTUR-con的AUC都达到0.64左右

**Comment by qq31415926** [comment]
验证集中有的用户没有出现在训练集中，这些用户的embedding是随机初始化的吗？

**Comment by nnnyt** [comment]
在我最终的实现里，对于LSTUR-ini，embedding是随机初始化的，对于LSTUR-con，embedding初始化为全0。具体实现可以查看代码https://github.com/nnnyt/NewsRecommendation/blob/e7fffb99b0e63f573be8c79be4984bd7bdb81b2d/models/LSTUR/model.py#L62-L69 印象中如果LSTUR-con的user embedding随机初始化的话效果不太好，但时间比较久了可能记得不太准确，你可以尝试一下

**Comment by qq31415926** [comment]
我现在做一个用bert优化LSTUR模型的实验，结果在small验证集上auc才61点多，未知用户我用最后一个user embedding初始化，不知道问题出在哪

**Comment by nnnyt** [comment]
用最后一个user embedding初始化新用户可能不是很合理？你可以再试试随机或者全0初始化😋或者再检查下其他细节

**Comment by qq31415926** [comment]
在lstur的model.py中的
`
        user_long_r = F.dropout2d(self.user_embedding(user).squeeze(dim=1).unsqueeze(dim=0), 
                                p=self.config.masking_probability, training=True).squeeze(dim=0)
`
squeeze(dim=1)是不是应该去掉，既然已经通过user embed为何还要进行squeeze操作？

**Comment by nnnyt** [comment]
是的，应该不需要这个squeeze。不过第1维的size不是1，`squeeze(dim=1)`其实没有作用，所以没有影响。

**Comment by qq31415926** [comment]
可以提供LSTUR.py中的这几个数据吗？
`
    news_index = np.load('news/news_index.npy', allow_pickle=True).item()
    news_index_test = np.load('news/news_index_test.npy', allow_pickle=True).item()
    word_index = np.load('news/word_index.npy', allow_pickle=True).item()
    news_title = np.load('news/news_title.npy', allow_pickle=True)
    news_category = np.load('news/news_category.npy', allow_pickle=True)
    news_subcategory = np.load('news/news_subcategory.npy', allow_pickle=True)
    all_news_test = np.load('news/all_news_test.npy', allow_pickle=True)
`

**Comment by nnnyt** [comment]
时间太久我已经删掉了😅你可以看一下我另一个[keras版本的仓库](https://github.com/nnnyt/MIND) 是把那个代码的预处理结果存下来直接使用的

**Comment by qq31415926** [comment]
哦好的，感谢🙏

**Comment by qq31415926** [comment]
您好，我还想问一下为啥config.py里的user_num要设置成56161+1，我统计了一下训练集中的behaviors.tsv的user数量只有50000

**Comment by nnnyt** [comment]
MIND数据在我做实验之后有更新过一次 可能有所变化 印象中数据格式好像也稍有差异来着
